src/core/modules/Planet.cpp

```diff
@@ -682,43 +683,53 @@ QString Planet::getInfoString(const StelCore* core, const InfoStringGroup& flags
 			{
 				// For compute the Moon age we use geocentric coordinates
 				QString moonPhase = "";
-				double eclJDE = earth->getRotObliquity(core->getJDE());
-				double ra_equ, dec_equ, lambdaMoon, lambdaSun, beta;
-				StelCore* core1 = StelApp::getInstance().getCore();
-				bool state = core1->getUseTopocentricCoordinates();
+				StelCore* core1 = StelApp::getInstance().getCore(); // FIXME: Why do we need a different reference to the (singular) core here?
+				const bool state = core1->getUseTopocentricCoordinates();
 				core1->setUseTopocentricCoordinates(false);
-				core1->update(0); // enforce update cache to avoid error in computation!
+				core1->update(0); // enforce update cache!
+				const double eclJDE = earth->getRotObliquity(core1->getJDE());
+				double ra_equ, dec_equ, lambdaMoon, lambdaSun, betaMoon, betaSun, raSun, deSun;
 				StelUtils::rectToSphe(&ra_equ,&dec_equ, getEquinoxEquatorialPos(core1));
-				StelUtils::equToEcl(ra_equ, dec_equ, eclJDE, &lambdaMoon, &beta);
-				StelUtils::rectToSphe(&ra_equ,&dec_equ, ssystem->getSun()->getEquinoxEquatorialPos(core1));
-				StelUtils::equToEcl(ra_equ, dec_equ, eclJDE, &lambdaSun, &beta);
-				double deltaLong = (lambdaMoon - lambdaSun)*M_180_PI;
+				StelUtils::equToEcl(ra_equ, dec_equ, eclJDE, &lambdaMoon, &betaMoon);
+				StelUtils::rectToSphe(&raSun,&deSun, ssystem->getSun()->getEquinoxEquatorialPos(core1));
+				StelUtils::equToEcl(raSun, deSun, eclJDE, &lambdaSun, &betaSun);
 				core1->setUseTopocentricCoordinates(state);
 				core1->update(0); // enforce update cache for avoid odd selection of Moon details!
+				double deltaLong = (lambdaMoon-lambdaSun)*M_180_PI;
 				if (deltaLong<0.) deltaLong += 360.;
-				int deltaLongI = qRound(deltaLong);
-				if (deltaLongI==45)
+				const int deltaLongI = static_cast<int>(std::round(deltaLong));
+				if (deltaLongI==0 || deltaLongI==360)
+					moonPhase = qc_("New Moon", "Moon phase");
+				else if (deltaLongI<90)
 					moonPhase = qc_("Waxing Crescent", "Moon phase");
-				if (deltaLongI==90)
+				else if (deltaLongI==90)
 					moonPhase = qc_("First Quarter", "Moon phase");
-				if (deltaLongI==135)
+				else if (deltaLongI<180)
 					moonPhase = qc_("Waxing Gibbous", "Moon phase");
-				if (deltaLongI==180)
+				else if (deltaLongI==180)
 					moonPhase = qc_("Full Moon", "Moon phase");
-				if (deltaLongI==225)
+				else if (deltaLongI<270)
 					moonPhase = qc_("Waning Gibbous", "Moon phase");
-				if (deltaLongI==270)
+				else if (deltaLongI==270)
 					moonPhase = qc_("Third Quarter", "Moon phase");
-				if (deltaLongI==315)
+				else if (deltaLongI<360)
 					moonPhase = qc_("Waning Crescent", "Moon phase");
-				if (deltaLongI==0 || deltaLongI==360)
-					moonPhase = qc_("New Moon", "Moon phase");
+				else
+					moonPhase = qc_("ERROR IN PHASE STRING PROGRAMMING!", "Moon phase");
```

Comment:
This is should be `qWarning()` - not translatable string for users

Reply:
Right. Rather meant for the developers, who should never see this. I must have been tired...